src/ocamldep.ml

```diff
@@ -75,6 +75,44 @@ module Dep_graphs = struct
 
   let wrapped_compat ~modules ~wrapped_compat =
     Ml_kind.Dict.make_both (Dep_graph.wrapped_compat ~modules ~wrapped_compat)
+
+  let merge_for_impl ~(vlib : t) ~(impl : t) =
+    { Ml_kind.Dict.
+      impl =
+        { Dep_graph.
+          dir = impl.impl.dir
+        ; per_module =
+            Module.Name.Map.merge vlib.impl.per_module impl.impl.per_module
+              ~f:(fun _ vlib impl ->
+                match vlib, impl with
+                | None, None -> assert false
+                | Some d, None
+                | None, Some d -> Some d
+                | Some v, Some i ->
```

Comment:
When can this happen?

Reply:
Oh, that's right. It can't happen since a module canont have more than 1 implementation.

Reply:
Actually, this can happen when the alias module is virtual. I think that it will still have an implementation (the generated one). I can handle this case a little more transparently.

Reply:
The alias module cannot be virtual, it's always generated by dune.

Reply:
I mean the lib_interface_module

Reply:
See the `variants-simple` example. Without this line, it will fail. It's not really an alias module however, as we special case such wrapped libraries with only 1 module.

Reply:
Ok, that makes sense. The comment bellow needs to be updated then